pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -51,14 +49,25 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, prox
 	clusters := make([]*v2.Cluster, 0)
 
 	services := push.Services
+	recomputeOutboundClusters := true
 
-	clusters = append(clusters, configgen.buildOutboundClusters(env, proxy, push, services)...)
-	for _, c := range clusters {
-		// Envoy requires a non-zero connect timeout
-		if c.ConnectTimeout == 0 {
-			c.ConnectTimeout = defaultClusterConnectTimeout
+	switch proxy.Type {
```

Comment:
I would keep the outbound clusters in PushContext.

I'm not sure how the cache is reset on config change - but I would rather not deal with 2 ways, one on the configen and one for the other configs.  We will want to do selective updates ( i.e. update only what changed, etc).